File: app/code-examples/credentials/credentials.py
with Client() as client:

    # Get Access token    
    response = client.post(
        url="https://auth.openepi.io/realms/openepi/protocol/openid-connect/token",
        data={
            "grant_type": "client_credentials",
            "client_id": "<YOUR_CLIENT_ID>",
            "client_secret": "<YOUR_CLIENT_SECRET>",
        },
    )

    
    token = response.json()['access_token']

    # Use access token to get data
    response = client.get(
        url="https://api-dev1.openepi.io/weather/locationforecast",
        params={"lat": 52.520008, "lon": 13.404954},
        headers={"Authorization": f"Bearer {token}"},
    )
    json = response.json()
    air_temperature = json["properties"]["timeseries"][0]["data"]["instant"]["details"]["air_temperature"]

    print(f"Air temperature: {air_temperature}")

File: app/code-examples/credentials/credentials.js
const tokenResponse = await fetch(
	"https://auth.openepi.io/realms/openepi/protocol/openid-connect/token",
	{
		method: "POST",
		body: JSON.stringify({
			grant_type: "client_credentials",
			client_id: "<YOUR_CLIENT_ID>",
			client_secret: "<YOUR_CLIENT_SECRET>",
		}),
	}
)
const tokenJson = await tokenResponse.json()
const token = tokenJson.access_token

// Use access token to get data

const weatherResponse = await fetch(
	`https://api.openepi.io/weather/locationforecast?lat=52.520008&lon=13.404954`,
	{
		headers: { Authorization: `Bearer ${token}` },
	}
)
const weatherJson = await weatherResponse.json()
const airTemperature =
	weatherJson.properties.timeseries[0].data.instant.details.air_temperature
console.log(`Air temperature: ${airTemperature}`)
